spotbugs-tests/src/test/java/edu/umd/cs/findbugs/detect/IncorrectSelfComparisonInstanceOfPatternMatchingTest.java

```diff
@@ -1,37 +1,32 @@
 package edu.umd.cs.findbugs.detect;
 
-import static org.hamcrest.core.Is.is;
-import static org.hamcrest.number.OrderingComparison.greaterThanOrEqualTo;
 import static org.hamcrest.MatcherAssert.assertThat;
-import static org.junit.Assume.assumeFalse;
-import static org.junit.Assume.assumeThat;
 
 import static edu.umd.cs.findbugs.test.CountMatcher.containsExactly;
 
 import java.nio.file.Paths;
 
-import org.junit.Rule;
-import org.junit.Test;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.condition.DisabledOnJre;
+import org.junit.jupiter.api.condition.JRE;
+import org.junit.jupiter.api.extension.ExtendWith;
 
 import edu.umd.cs.findbugs.BugCollection;
-import edu.umd.cs.findbugs.test.SpotBugsRule;
+import edu.umd.cs.findbugs.test.SpotBugsExtension;
+import edu.umd.cs.findbugs.test.SpotBugsRunner;
 import edu.umd.cs.findbugs.test.matcher.BugInstanceMatcher;
 import edu.umd.cs.findbugs.test.matcher.BugInstanceMatcherBuilder;
 
-public class IncorrectSelfComparisonInstanceOfPatternMatchingTest {
-    @Rule
-    public SpotBugsRule spotbugs = new SpotBugsRule();
+@ExtendWith(SpotBugsExtension.class)
+class IncorrectSelfComparisonInstanceOfPatternMatchingTest {
 
     /**
      * @see <a href="https://github.com/spotbugs/spotbugs/issues/1136">GitHub
      *      issue</a>
      */
     @Test
-    public void testIssue1136() {
-        assumeFalse(System.getProperty("java.specification.version").startsWith("1."));
-        int javaVersion = Integer.parseInt(System.getProperty("java.specification.version"));
-        assumeThat(javaVersion, is(greaterThanOrEqualTo(14)));
-
+    @DisabledOnJre(JRE.JAVA_11)
```

Comment:
Yes because we only build (in future pr) with jdk 11, 17, or 21.  Thus we don't need to mention any other jdks.  So we only need to exclude it against jdk 11 builds.

Reply:
We could add other JDKs later to build, so let's not loose this info, it would be much harder later to figure it out. Also, it's about the JRE, not the JDK.